core-sdk-samples/higgs-shop-sample-app/README.md

```diff
@@ -21,25 +21,116 @@ By using this sample app, you will quickly learn how to:
+        total_product_amounts: subTotal,
+    });
+}, []);
+```
+
+In some cases, we fire a _Commerce Event_ instead of a _Page View_ to track more e-Commerce related attributes.
+
+### Custom Events
+
+Most often, you will need to use [Custom Events](https://docs.mparticle.com/developers/sdk/web/event-tracking/#custom-events) to track events in a way that is unique to your use case. mParticle provides types of _Custom Events_ ranging from Navigation Events to Social Media Engagement and are mostly used to organize your data in a way that makes sense to you.
+
+Many of our components in `/src/components` make use of these events, particularly the `NavigationMenuItem` Component.
+
+```typescript
+mParticle.logEvent(
+    'Navbar Click',
+    mParticle.EventType.Navigation,
+    customAttributes,
+);
+```
+
+### Commerce Events
+
+This Sample App emulates a simple e-Commerce application and makes heavy use of mParticle's [Commerce Events](https://docs.mparticle.com/developers/sdk/web/commerce-tracking/).
+
+Some events used in this application:
+
+-   Add To Cart
+-   Remove From Cart
+-   Product Detail
+-   Product Impression
+-   Checkout
+-   Purchase
+
+Most _Commerce Events_ follow a similar pattern, requiring that you first generate an **mParticle Product** Object, which then gets passed into the `logProductAction` method.
+
+You should map your own product attributes to be consistent with your [Data Plan](https://docs.mparticle.com/guides/data-master/introduction/) if you are leveraging that feature. Using Data Plans ensures data consistency within an app and across devices.
+
+```typescript
+const { label, id, price, color, size } = product;
+
+const mParticleProduct: mParticle.Product = mParticle.eCommerce.createProduct(
+    label,
+    sku,
+    price,
+    quantity,
+);
+
+mParticle.eCommerce.logProductAction(
+    mParticle.ProductActionType.AddToCart,
+    mParticleProduct,
+);
+```
+
+Most Commerce Events are used within the following components:
+
+-   `/src/pages/ProductPage.tsx`
+-   `/src/pages/CartPage.tsx`
+-   `/src/features/OrderDetails/OrderDetailsCartView`
+-   `/src/features/OrderDetails/OrderDetailsPurchaseReview`
+
+## Discovering Events
+
+As a developer, sometimes the best way to learn is to just dig into the code or your debugging tools. To that end, this sample app ships with a verbose logger that you can view details of what our SDK is doing within your browser's console.
+
+### Browser Console
+
+![mParticle Console Logs](./.github/images/logger.jpg)
+
+![mParticle Network Events](./.github/images/event-request.jpg)
+
+## Explore an example an end-to-end data flow with HiggsMart and mParticle
+mParticle simplifies the process of collecting customer data events from your apps, and forwarding this data to a wide variety of downstream systems for analytics, data processing, and marketing. To illustrate this process, here is an example of a data connection using HiggsMart as an input, and a sample output. 
```

Comment:
I think we should add the following to avoid having to deal with any support requests in case Webhook.site is having any issues.

> In this example, we will be using [Webhook.site](https://www.webhook.site/) to simulate data transferring between the Sample App, and your own services.
>
> Webhook.site is not an mParticle service and is only being used as to simulate an end-to-end full data flow to a third party service.
```